### PR TITLE
GG-20249 Extract message from exception cause stack trace.

### PR DESCRIPTION
**Pull request by vsisko**

### CODE REVIEW

File: modules/web-console/frontend/app/services/ErrorParser.service.js [hunk 1]
```diff
@@ -17,6 +17,8 @@
 import isEmpty from 'lodash/isEmpty';
 import {nonEmpty} from 'app/utils/lodashMixins';
 
+const causeStr = 'Caused by: ';
```
Comment (akuznetsov-gridgain) [inline modules/web-console/frontend/app/services/ErrorParser.service.js]: Use UPPER_CASE instead.

File: modules/web-console/frontend/app/services/Messages.service.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default function factory($alert, errorParser) {
 
         const title = err ? errorMessage(message, err) : errorMessage(null, message);
 
-        msgModal = $alert({type, title, duration});
+        msgModal = $alert({type, title, duration: duration + ((title.match(/<li>/g) || []).length * 3)});
```
Comment (akuznetsov-gridgain) [inline modules/web-console/frontend/app/services/Messages.service.js]: Lets think about better solution. I do not like regex parsing.

File: modules/web-console/frontend/app/services/ErrorParser.service.js (hunk 1 above)
Comment (vsisko) [reply modules/web-console/frontend/app/services/ErrorParser.service.js]: Fixed constant name.

File: modules/web-console/frontend/app/services/Messages.service.js (hunk 2 above)
Comment (vsisko) [reply modules/web-console/frontend/app/services/Messages.service.js]: Implemented returning of metadata information of error parsing.

File: modules/web-console/frontend/app/services/ErrorParser.service.ts [hunk 3]
```diff
@@ -58,20 +60,39 @@ export default class {
...
+                    const msgEnd = err.message.indexOf('[', msgStart);
+                    const cause = err.message.substring(msgStart, msgEnd >= 0 && msgEnd < causeEndLine ? msgEnd : causeEndLine);
+
+                    causes.unshift(cause);
+
+                    causeIdx = err.message.indexOf(CAUSE_STR, causeIdx + CAUSE_STR.length);
+                }
+
+                return new ErrorParseResult(
+                    prefix + (lastIdx >= 0 ? msg.substring(lastIdx + 5, msgEndIdx > 0 ? msgEndIdx : traceIndex) : msg)
```
Comment (akuznetsov-gridgain) [inline modules/web-console/frontend/app/services/ErrorParser.service.ts]: Please refactor to store causes in ErrorParseResult.

**Pull request merged**